src/integrations.py

```diff
@@ -335,23 +338,31 @@ class TLSCertificates:
     ca_bundle: str
 
     @classmethod
-    def load(cls, requirer: CertificateTransferRequires) -> "TLSCertificates":
+    def load(
+        cls, requirer: CertificateTransferRequires | V1CertificateTransferRequires
+    ) -> "TLSCertificates":
         """Fetch the CA certificates from all "receive-ca-cert" integrations.
 
         Compose the trusted CA certificates in /etc/ssl/certs/ca-certificates.crt.
         """
+        if isinstance(requirer, V1CertificateTransferRequires):
+            return cls(ca_bundle="\n".join(requirer.get_all_certificates()))
```

Comment:
We should not return here, if we relate the charm with self-signed-certificates the requirer object will still be `V1CertificateTransferRequires`, but the `requirer.get_all_certificates` will return an empty string.

Reply:
Just another question (I probably lack some contexts or misunderstand something):

Can't the following `for` loop cover both cases, no matter whether `v0` or `v1` is used?

Reply:
`v1` fetches the `certificates` key from the databag, while `v0` fetches the `ca` key. Not sure if it's a pure rename or there differences in what is stored in there.

`v1` comes with a helper function (which is what is used here), while v0 has no helper functions, thus the current approach. I think it is best to use existing functions where possible.